pkg/adapter/loki/lokiLogsProvider.go

```diff
@@ -0,0 +1,252 @@
+package loki
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	col "github.com/cloudogu/k8s-support-archive-operator/pkg/adapter/collector"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const loggerName = "LokiLogsProvider"
+const maxQueryTimeWindowInDays = 30 // Loki's max time range is 30d1h
+const maxQueryResultCount = 2000    // Loki's max entries limit per query is 5000
+
+type LokiLogsProvider struct {
+	serviceURL string
+	httpClient *http.Client
+	username   string
+	password   string
+}
+
+type labelValuesResponse struct {
+	Status string   `json:"status"`
+	Data   []string `json:"data"`
+}
+
+type queryLogsResponse struct {
+	Status string        `json:"status"`
+	Data   queryLogsData `json:"data"`
+}
+
+type queryLogsData struct {
+	ResultType string            `json:"resultType"`
+	Result     []queryLogsResult `json:"result"`
+}
+
+type queryLogsResult struct {
+	Stream queryLogsStream
+	Values [][]string `json:"values"`
+}
+
+type queryLogsStream struct {
+	LogLevel  string `json:"detected_level"`
+	Namespace string `json:"namespace"`
+	Kind      string `json:"kind"`
+}
+
+func NewLokiLogsProvider(httpClient *http.Client, httpAPIUrl string, username string, password string) *LokiLogsProvider {
+	return &LokiLogsProvider{
+		serviceURL: httpAPIUrl,
+		httpClient: httpClient,
+		username:   username,
+		password:   password,
+	}
+}
+
+func (lp *LokiLogsProvider) FindLogs(
+	ctx context.Context,
+	startTimeInNanoSec int64,
+	endTimeInNanoSec int64,
+	namespace string,
+	resultChan chan<- *col.LogLine,
+) error {
+	var reqStartTime, reqEndTime = int64(0), startTimeInNanoSec
+	for {
+		reqStartTime, reqEndTime = findLogsNextTimeWindow(reqEndTime, endTimeInNanoSec, maxQueryTimeWindowInDays)
+		httpResp, err := lp.httpFindLogs(ctx, reqStartTime, reqEndTime, namespace)
+		if err != nil {
+			return fmt.Errorf("find logs; %v", err)
+		}
+
+		logLines, err := convertQueryLogsResponseToLogLines(httpResp)
+		if err != nil {
+			return fmt.Errorf("convert http response to LogLines; %v", err)
+		}
+		if len(logLines) == 0 {
+			return nil
+		}
+
+		for _, ll := range logLines {
+			resultChan <- &ll
+		}
```

Comment:
Why are you doing this here instead of using the last end timestamp as new start?
There might be a good reason. A comment would be nice here.

At some point this call porduces an infinite loop because the timestamps are always equal.